### PR TITLE
3 task routing

### PR DESCRIPTION
**Pull request by otecpavlo**
1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/modules-services-routing.md
2. Done 19.08.2021 / deadline 16.08.2021
:heavy_check_mark: The 404 page and redirection logic is implemented
:heavy_check_mark: The Login page functionality with necessary rules is implemented
:heavy_check_mark: The Detailed information page is implemented
:heavy_check_mark: The app.component doesn't include any logic and used only for markup
:heavy_check_mark: At least two services (Login service and Youtube service) are implemented

### CODE REVIEW

File: youtube-client/src/app/auth/components/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Component({
+  selector: 'login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+
+  constructor(private authService: AuthService, private router: Router) { }
+
+  ngOnInit(): void {
```
Comment (4m9fk) [inline youtube-client/src/app/auth/components/login/login.component.ts:14]: Why do we need those empty ngOnInit's?

File: youtube-client/src/app/youtube/components/search-result-item/search-result-item.component.ts [hunk 2]
```diff
@@ -9,9 +10,12 @@ import { ISearchItem } from '../models/search-result-item.model';
...
   @Input() item?: ISearchItem;
 
-  constructor() { }
+  constructor(private router: Router) { }
 
   ngOnInit(): void {
   }
 
+  itemDescription() {
+    this.router.navigate([`description/${this.item?.id}`]);
```
Comment (4m9fk) [inline youtube-client/src/app/youtube/components/search-result-item/search-result-item.component.ts:19]: this.item?.id
Is it possible to have an item without id?

Comment (4m9fk) [reply youtube-client/src/app/youtube/components/search-result-item/search-result-item.component.ts:19]: Also, please find a better name for `itemDescription`

Comment (otecpavlo) [reply youtube-client/src/app/youtube/components/search-result-item/search-result-item.component.ts:19]: - added condition `if (!this.item) return`; 
- `itemDescription` renamed to `navigateDetailedPage`

File: youtube-client/src/app/auth/components/login/login.component.ts (hunk 1 above)
Comment (otecpavlo) [reply youtube-client/src/app/auth/components/login/login.component.ts:14]: removed all empty ngOnInit's and constructors

**Pull request merged**